### PR TITLE
Migrate to new `py2exe` build API

### PR DESCRIPTION
**Pull request by Athanasius**
Due to deprecation, and  removal in Python 3.12, of `distutils`, py2exe has had to implement a new API for building using `setuptools`.

This PR introduces a new file `Build-exe-and-msi.py` which utilises the new method.  There are actually minimal changes necessary, but this way we have the old method available, for now, in `setup.py` without introducing a command-line switch or the like.  The GitHub workflow has been updated to use the new method, and `docs/Releasing.md` has also been updated.

This also removes the macOS packaging as we've not used it since the maintainership change in 2020, it's still in `setup.py` though.

### CODE REVIEW

File: Build-exe-and-msi.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+if sys.version_info[0:2] != (3, 10):
+    raise AssertionError(f'Unexpected python version {sys.version}')
+
+if sys.platform == 'win32':
+    assert platform.architecture()[0] == '32bit', 'Assumes a Python built for 32bit'
+    import py2exe  # noqa: F401 # Yes, this *is* used
+    dist_dir = 'dist.win32'
+
+else:
+    assert False, f'Unsupported platform {sys.platform}'
```
Comment (A-UNDERSCORE-D) [inline Build-exe-and-msi.py:30]: May as well just `raise AssertionError()` here if you want specifically that exception

File: Build-exe-and-msi.py [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    options=OPTIONS,
+)
+###########################################################################
+
+###########################################################################
+# Build installer(s)
+###########################################################################
+package_filename = None
+# Ensure this fails on non-win32
+if sys.platform == 'win32':
```
Comment (A-UNDERSCORE-D) [inline Build-exe-and-msi.py:176]:
```suggestion
if sys.platform != 'win32':
    raise AssertionError(...)
```

Or `assert platform == 'win32'`?

**Review by A-UNDERSCORE-D — COMMENTED** [review]
Couple of changes I'd make.

Also is there a particular reason this starts with a capital letter? :D

**Comment by Athanasius** [comment]
> Also is there a particular reason this starts with a capital letter? :D

Oh, you mean the file name.  Yeah, I wanted it to sort differently, and stand out from "actual application code".  Not that it helps Windows 'git bash' tab-completion not confuse it with the `build/` directory it causes to exist.

File: Build-exe-and-msi.py (hunk 1 above)
Comment (Athanasius) [reply Build-exe-and-msi.py]: Changes to use that.

File: Build-exe-and-msi.py (hunk 2 above)
Comment (Athanasius) [reply Build-exe-and-msi.py]: I've removed all the `win32` tests except the first one now.  I was too much in "minimal changes" mindset the other day.

**Comment by A-UNDERSCORE-D** [comment]
> > Also is there a particular reason this starts with a capital letter? :D
> 
> Oh, you mean the file name.  Yeah, I wanted it to sort differently, and stand out from "actual application code".  Not that it helps Windows 'git bash' tab-completion not confuse it with the `build/` directory it causes to exist.

Ah okay

**Review by A-UNDERSCORE-D — APPROVED** [review]

**Pull request merged**